core/controllers/questions_list.py

```diff
@@ -61,12 +65,18 @@ def get(self, comma_separated_skill_ids):
         except Exception as e:
             raise self.PageNotFoundException(e)
 
-        (
-            question_summaries, merged_question_skill_links,
-            next_start_cursor) = (
-                question_services.get_displayable_question_skill_link_details(
-                    constants.NUM_QUESTIONS_PER_PAGE, skill_ids, start_cursor)
-            )
+        question_summaries, merged_question_skill_links, _ = (
+            question_services.get_displayable_question_skill_link_details(
+                constants.NUM_QUESTIONS_PER_PAGE + 1, skill_ids, offset)
+        )
+
+        if len(question_summaries) <= constants.NUM_QUESTIONS_PER_PAGE:
+            more = False
+        else:
+            more = True
+            question_summaries.pop()
+            merged_question_skill_links.pop()
```

Comment:
Doesn't `pop` remove only one item? If `len(question_summaries)` was 10 and `NUM_QUESTIONS_PER_PAGE` is 5, wouldn't the length be 9 after this change and so still not valid?

Reply:
I am fetching only `NUM_QUESTIONS_PER_PAGE + 1`. So in this case `len(question_summaries)` can be 6 or less. And if it is 6, pop() is called.